pandas/io/excel/_base.py

```diff
@@ -204,6 +204,9 @@
     Detect missing value markers (empty strings and the value of na_values). In
     data without any NAs, passing na_filter=False can improve the performance
     of reading a large file.
+skip_blank_lines : bool, default True
+    If True, skip over blank lines in single-column spreadsheets rather than
+    interpreting as NaN values.
```

Comment:
What is the behavior when there is more than one column?

Reply:
When there are multiple columns, blank rows are never skipped. The 
discussion on #39808 has more detail on why this happens and whether we should do something different. Advice welcome.

Reply:
I see - thanks. We shouldn't have user code that reads:

```
df = pd.read_excel("test.xlsx", skip_blank_lines=True)
```

that results in not skipping blank lines. Instead of passing on to TextParser, can you implement as a method on BaseExcelReader and then call after get_sheet_data?

Reply:
I agree that the current version isn't right, but I'm not sure that skipping fully-empty rows is right either because it wouldn't match the CSV behaviour. `read_csv(skip_blank_lines=True)` skips completely blank lines (`\n`) but it doesn't skip lines of empty cells (`""\n` or `,,,\n`).

I'm starting to think that "blank lines" are a problem unique to CSVs and read_excel shouldn't have this argument at all; it should simply call TextParser with skip_blank_lines=False. That would solve issue #39808 (can't turn off the single-column line skipping) and make the read_excel behaviour more similar to read_excel.

Would that be a reasonable solution?

Reply:
I think that solves the most pressing issue, and certainly +1 on such a PR. But the OP of #39808 also mentions not being able to pass `skip_blank_lines=True` (see the 3rd SO referenced there). I think of a blank line in an excel file as "a row with no data in any of the columns" and a blank line in a CSV as "a line with no characters on it". So IMO, it is okay to not "match" behavior. But the docs of read_csv and read_excel should also call this peculiarity out, if implemented.

The only reason why I find adding `skip_blank_lines` compelling is because it used to be a feature but was (as far as I can tell) accidentally removed. Users can always remove the rows after reading.

cc @jreback @topper-123 @WillAyd for any thoughts here.

Reply:
I think that the PythonParser(engine the TextParser uses) should probably be modified(the _is_line_empty function to be precise https://github.com/pandas-dev/pandas/blob/9fdb8f695dcf6c38326236e94c4ab4fc27911dfd/pandas/io/parsers/python_parser.py#L596). I am also +1 on having the feature, but I think the bug should be fixed before adding back.

Reply:
Makes sense. I'll start by fixing the current bug. Expanding the definition of `skip_blank_lines` to include all empty rows could be desirable, but I'd want some more discussion to make sure we're all imagining the same behaviour and it's impact on existing code. For instance, if we kept the default to `True` as @WillAyd recommended in #39808, and if we filter out the blank lines between the calls to get_sheet_data and TextParser then we would break every call to read_excel that uses skiprows to skip empty rows.

I think it's cleanest to close this PR and start a new one for just the bugfix.

Reply:
That sounds great - I've also been rethinking my comment above. As I mentioned there, I don't find the use-case of `skip_blank_lines` compelling for excel files, agreeing with your previous comment on this above. And we certainly shouldn't bring back the ability just to then deprecate it properly.